src/pages/api/logs/[id].js

```diff
@@ -0,0 +1,84 @@
+import mongoose, { Types } from "mongoose";
+import { updateLog } from "../../../../server/db/actions/Log";
+import { z } from "zod";
+import { consts } from "@/utils/consts";
+
+const logSchema = z.object({
+  title: z.string().optional(),
+  topic: z.enum(consts.topicArray).optional(),
+  tags: z.enum(consts.tagsArray).array().optional(),
+  severity: z.enum(consts.concernArray).optional(),
+  description: z.string().optional().optional(),
+});
+
+export default async function handler(req, res) {
+  if (req.method == "PATCH") {
+    if (!mongoose.isValidObjectId(req.query.id)) {
+      return res.status(422).send({
+        success: false,
+        message: "Unable to update because log ID is not in valid format.",
+      });
+    }
+    const { success, error, data } = logSchema
+      .partial()
+      .strict()
+      .safeParse(req.body);
+
+    if (!success) {
```

Comment:
when enum values are invalid (like `tags: ["fake tag"]`, there is a property undefined error on line 45. I think this may be because invalid enums have error code `invalid_enum_value`